src/rocker/core.py

```diff
@@ -338,14 +347,27 @@ def write_files(extensions, args_dict, target_directory):
 
 def generate_dockerfile(extensions, args_dict, base_image):
     dockerfile_str = ''
+    # Preamble snippets
     for el in extensions:
         dockerfile_str += '# Preamble from extension [%s]\n' % el.name
         dockerfile_str += el.get_preamble(args_dict) + '\n'
     dockerfile_str += '\nFROM %s\n' % base_image
+    # ROOT snippets
     dockerfile_str += 'USER root\n'
     for el in extensions:
         dockerfile_str += '# Snippet from extension [%s]\n' % el.name
         dockerfile_str += el.get_snippet(args_dict) + '\n'
+    # Set USER if user extension activated
+    if 'user' in args_dict and args_dict['user']:
+        if 'user_override_name' in args_dict and args_dict['user_override_name']:
+            username = args_dict['user_override_name']
+        else:
+            username = get_user_name()
+        dockerfile_str += f'USER {username}\n'
+    # USER snippets
+    for el in extensions:
```

Comment:
These user snippets will execute whether the `user` extension is present or not. Which I think is fine, however, some extensions might expect there to be a `/home/<user>` directory at this point. Is there a good way for those extensions to detect and handle that case?

I suppose one way (albeit cumbersome) is to add a `user_exists_preamble` and a `no_user_preamble` that gets called immediately before the `user_snippet`. This allows an extension to set up variables as needed to execute the user snippet as root or as a user.

Reply:
I realized since I posted this that the `get_user_snippet` method has the cli_args available. An extension should be able to determine if the `user` extension is enabled/disabled within this method call and can modify the docker return string as needed.